repobuild/rpm/gen-repo.sh

```diff
@@ -39,18 +39,23 @@ cp -r $dir/../../pkgbuild/SRPMS/*       $pool
 echo "%_gpg_name Assurio Software Package Developers <packages@assurio.com>" >> ~/.rpmmacros
 
 #Sign packages
-find $pool -type f -name "*.rpm" -exec $dir/rpm-sign.exp {} \;
+for f in `ls $pool/*.rpm`; do
+    $dir/rpm-sign.exp $f || die "Failed to sign rpm"
+done
 
 # Generate repodata/repomd.xml
-createrepo $cent6/x86_64
+createrepo $cent6/x86_64 || die "Failed to create repomd.xml"
 
 # Sign repository (repomd.xml files)
-gpg2 --default-key $GPG_KEY_ID --batch --yes --no-tty --armor --digest-algo SHA256 --detach-sign $cent6/x86_64/repodata/repomd.xml
+gpg2 --default-key $GPG_KEY_ID --batch --yes --no-tty --armor --digest-algo SHA256 --detach-sign $cent6/x86_64/repodata/repomd.xml || \
+    die "Failed to sign repomd.xml file."
 
 # Generate 'mirrors'
-mirrors_content="http://$bucket.s3.amazonaws.com/repo/assurio-snap/$branch/rpm/CentOS/\$releasever/\$basearch"
-for t in '' 'Client' 'Server' 'Workstation'; do
-	echo $mirrors_content > $out/CentOS/mirrors/6$t
+mirrors_content="http://$bucket.s3.amazonaws.com/repo/assurio-snap/$branch/rpm/CentOS/6/\$basearch"
```

Comment:
This should be `https` I think.

Reply:
Usually CentOS has 'http' links to their mirrors. For instance, you can find mirror list for the 'extras' repo here: http://mirrorlist.centos.org/?release=6&arch=x86_64&repo=extras
There are just http links. The same is for CentOS 7 and 8.
But I tried to use 'https' link per your suggestion and got this error:
```
https://ci.assur.io.s3.amazonaws.com/repo/assurio-snap/pkg-build/rpm/CentOS/6/x86_64/repodata/repomd.xml: [Errno 14] PYCURL ERROR 51 - "SSL: certificate subject name '*.s3.amazonaws.com' does not match target host name 'ci.assur.io.s3.amazonaws.com'"
```
And a browser warns about not secure connection when opening `https://ci.assur.io.s3.amazonaws.com/repo/assurio-snap/pkg-build/rpm/CentOS/6/x86_64/repodata/repomd.xml`

Reply:
upd:
https://forums.aws.amazon.com/message.jspa?messageID=207082 :
`The certificate match problem occurs with any buckets that containt "." characters. Unfortunately, *.s3.amazonaws.com is not equivalent to ..s3.amazonaws.com for SSL certs. `

And when using 'https://s3.amazonaws.com/ci.assur.io/repo....' getting this redirect issue:
```
YumRepo Error: All mirror URLs are not using ftp, http[s] or file.
 Eg. <Error><Code>PermanentRedirect</Code><Message>The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.</Message><Endpoint>ci.assur.io.s3.amazonaws.com</Endpoint><Bucket>ci.assur.io</Bucket><RequestId>B1AD75FEE29CA810</RequestId><HostId>dV1Xjd6seHVxae9PCLn6OKAedH7NT7zuevNTiYSVQnbexsD1uWZ7+yO4DQHqfOXAJpwaQ9SVJ10=</HostId></Error>/
removing mirrorlist with no valid mirrors: /var/cache/yum/x86_64/6/Assurio/mirrorlist.txt
```
Let's do a subdomain https://repo.assur.io/ or so.

Reply:
Updated PR with commit https://github.com/assuriosw/assurio-snap/pull/5/commits/19be1f00983dcb0bb42aafa3fc94a40131644e70